src/appendices/spring/thymeleaf.rst

```diff
@@ -0,0 +1,108 @@
+.. _thymeleaf-commands:
+
+Thymeleaf Commands
+==================
+
+Thymeleaf is a template engine and Thymeleaf templates make up the views of our MVC application.
+In order to create our templates, we use attributes to pass data through controllers to our views out of the model.
+
+Attributes
+----------
+
+Thymeleaf has many attributes that we could use. In this book, we focus on a few key ones.
+
+Displaying Data
+^^^^^^^^^^^^^^^
+
+``th:text`` dynamically populates the contents of an HTML element.
+
+.. sourcecode:: html
+
+   <p th:text = "Hello World!">text</p>
+
+If we want to pull in a value of a variable from the controller, we can use the **variable expressions** syntax, ``${}``.
+Let's say we want to pull in the value of a variable called ``hello``.
+
+.. sourcecode:: html
+
+   <p th:text = "${hello}">text</p>
+
+Conditionally Displaying Data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to conditionally display data, we need to use ``th:if`` and ``th:unless``.
+``th:if`` will display the value of the element if the expression evaluates to true.
+``th:unless`` will display the value of the element if the expression evaluates to false.
+
+.. admonition:: Example
+   
+   Let's say we want to conditionally display our grocery list. We do have a variable, ``pantryStatus``, that is a String.
+   If ``pantryStatus`` is ``empty``, we want to display a list that includes staples like flour, sugar, and rice.
+   If ``pantryStatus`` is not ``empty``, we want our grocery list to include our favorite fresh fruits and veggies, like bananas, strawberries, and broccoli.
+
+   .. sourcecode:: html
+      :linenos:
+
+      <ol id = "groceryList">
+         <li th:if = "${pantryStatus.equals("empty")}">Flour</li>
+         <li th:if = "${pantryStatus.equals("empty")}">Sugar</li>
+         <li th:if = "${pantryStatus.equals("empty")}">Rice</li>
+         <li th:unless = "${pantryStatus.equals("empty")}">Bananas</li>
+         <li th:unless = "${pantryStatus.equals("empty")}">Strawberries</li>
+         <li th:unless = "${pantryStatus.equals("empty")}">Broccoli</li>
+      </ol>
+
```

Comment:
Similarly to ``*ngIf``, what do you think about adding a quick reminder about the difference between *adding* an HTML element vs. *displaying* it (or *hiding* vs. *removing*)?

Reply:
I think this info should go into the chapter!

Reply:
Got it.  I'll add the reminder to the Thymeleaf conditionals page.